### PR TITLE
chore: remove getProjectExperiments and merge into getExperiments

### PR DESCRIPTION
**Pull request by mapmeld**
## Description

Separated out from the experiment.state work in #4420 , this removes the separate `getProjectExperiments` API endpoint and combines it with `getExperiments`.  If we're going to change the API it makes sense to package that change on its own and roll it out sooner rather than later.

## Test Plan

Project pages and tests should run normally

## Checklist

- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.
- [x] If modifying `/webui/react/src/shared/` verify `make -C webui/react test-shared` passes.

### CODE REVIEW

File: docs/release-notes/4606-get-experiments-consolidation.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+:orphan:
+
+**Breaking Changes**
+
+-  API: the `GetProjectExperiments` endpoint (`/projects/:id/experiments`) is removed and replaced
```
Comment (gh-determined-ai) [inline docs/release-notes/4606-get-experiments-consolidation.txt:5]: I believe we're using double backticks (``), not single (`) in our docs.

**Review by gh-determined-ai — APPROVED** [review]

File: docs/release-notes/4606-get-experiments-consolidation.txt (hunk 1 above)
Comment (mapmeld) [reply docs/release-notes/4606-get-experiments-consolidation.txt]: Pushed new version with ``

**Review by EmilyBonar — APPROVED** [review]
LGTM! It's probably outside the scope of this PR, but would it be possible for `getExperiments` to also return the experiment configs (i.e. returning `ExperimentBase` vs `ProjectExperiment` or `ExperimentItem`)? That would allow us to have HP Search as an option in the context menu on the Project Details page.

**Review by ioga — APPROVED** [review]

**Comment by mapmeld** [comment]
@ioga - added **config** object to GetExperiments and moved it from a separate attribute to part of the experiment in GetExperiment; lemme know if this raises issues.

**Review by ioga — APPROVED** [review]
approved, but let's try to call next version 0.19.0

**Pull request merged**